### PR TITLE
[SYCL][ESIMD][E2E] Fix esimd emulator stack overflow

### PR DESCRIPTION
**Pull request by sarnex**
With the code running on the host, the #pragma unrolls cause absolutely massive code that ends up overflowing the stack. Don't explicitly unroll for the emulator.

### CODE REVIEW

**Comment by sarnex** [comment]
```

********************
Failed Tests (1):
  SYCL :: Plugin/level_zero_imm_cmdlist_per_thread.cpp
```

not related

**Review by v-klochkov — COMMENTED** [review]
I have only couple of non-critical comments.

File: sycl/test-e2e/ESIMD/BitonicSortK_emu.cpp [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: esimd_emulator
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+#define ESIMD_EMU
+
+#include "BitonicSortK.hpp"
```
[the remark this answers is not in the document]
Comment (sarnex) [reply sycl/test-e2e/ESIMD/BitonicSortK_emu.cpp:14]: usually the pattern i see is the hpp file is completely generic and requires defines to compile/run, and the .cpp file has the LIT commands and provides the defines

File: sycl/test-e2e/ESIMD/BitonicSortK.hpp [hunk 2]
```diff
@@ -0,0 +1,725 @@
...
+
+#include <algorithm>
+#include <iostream>
+#include <sycl/ext/intel/esimd.hpp>
+#include <sycl/sycl.hpp>
+
+// Disable unrolling with the esmid emulator
+// due to huge stack size requirement
+#ifdef ESIMD_EMU
+#define ESIMD_UNROLL
```
Comment (v-klochkov) [inline sycl/test-e2e/ESIMD/BitonicSortK.hpp:19]: wouldn't it be safer than to say: "nounroll" for emulator?

**Review by v-klochkov — APPROVED** [review]

**Comment by v-klochkov** [comment]
CUDA fails are unrelated to this PR.

**Pull request merged**